DDDEastAnglia.Tests/InMemoryRepository.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace DDDEastAnglia.Tests
+{
+    internal class InMemoryRepository<T>
+    {
+        private readonly Func<T, int> keyFunc;
+        private readonly ConcurrentDictionary<int, T> store = new ConcurrentDictionary<int, T>();
+
+        public InMemoryRepository(Func<T, int> keyFunc)
```

Comment:
Mmm good question.. not sure.. I didn't want to introduce a load of boiler plate classes for this so a simple delegate seemed more appropriate.. I could have a named delegate? or use an alias? 

It's used to tell the generic class where the id is. .for get by id calls. 

As we're using SQL.. our primary keys will be ints I'm assuming.

how about `primarykeyfinder`?